aggregators/aggregator.go

```diff
@@ -603,23 +603,133 @@ func (a *Aggregator) processHarvest(
+						"If you are using an agent with 'use_path_as_transaction_name' enabled, it may cause "+
+						"high cardinality. If your agent supports the 'transaction_name_groups' option, setting "+
+						"that configuration option appropriately, may lead to better results.", overflowBucketName),
+						zap.Duration("aggregation_interval_ns", aggIvl),
+						zap.Int("limit", a.cfg.Limits.MaxTransactionGroups),
+					)
+					txOverflowLogged = true
+				}
+			}
+		}
+
+		if overflowed := hllSketchEstimate(o.OverflowServiceTransactionsEstimator); overflowed > 0 {
+			hs.serviceTransactionsOverflowed += overflowed
+			if serviceName != overflowBucketName {
+				if svcTxGroups >= a.cfg.Limits.MaxServiceTransactionGroupsPerService {
+					logFunc(fmt.Sprintf(""+
+						"Service transaction group per service limit reached, new metric documents will be grouped "+
+						"under a dedicated bucket identified by transaction type '%s'.", overflowBucketName),
+						zap.String("service_name", serviceName),
+						zap.Duration("aggregation_interval_ns", aggIvl),
+						zap.Int("limit", a.cfg.Limits.MaxServiceTransactionGroupsPerService),
+					)
+				} else if !svcTxOverflowLogged {
+					logFunc(fmt.Sprintf(""+
+						"Overall service transaction group limit reached, new metric documents will be grouped "+
+						"under a dedicated bucket identified by transaction type '%s'.", overflowBucketName),
+						zap.Duration("aggregation_interval_ns", aggIvl),
+						zap.Int("limit", a.cfg.Limits.MaxServiceTransactionGroups),
+					)
+					svcTxOverflowLogged = true
+				}
+			}
+		}
+
+		if overflowed := hllSketchEstimate(o.OverflowSpansEstimator); overflowed > 0 {
+			hs.spansOverflowed += overflowed
+			if serviceName != overflowBucketName {
+				if spanGroups >= a.cfg.Limits.MaxSpanGroupsPerService {
+					logFunc(fmt.Sprintf(""+
+						"Span group per service limit reached, new metric documents will be grouped "+
+						"under a dedicated bucket identified by service target name '%s'.", overflowBucketName),
+						zap.String("service_name", serviceName),
+						zap.Duration("aggregation_interval_ns", aggIvl),
+						zap.Int("limit", a.cfg.Limits.MaxSpanGroupsPerService),
+					)
+				} else if !spanOverflowLogged {
+					logFunc(fmt.Sprintf(""+
+						"Overall span group limit reached, new metric documents will be grouped "+
+						"under a dedicated bucket identified by service target name '%s'.", overflowBucketName),
+						zap.Duration("aggregation_interval_ns", aggIvl),
+						zap.Int("limit", a.cfg.Limits.MaxSpanGroups),
+					)
+					spanOverflowLogged = true
+				}
+			}
+		}
 	}
 
-	if err := a.cfg.Processor(ctx, cmk, cm, aggIvl); err != nil {
-		return harvestStats{}, fmt.Errorf("failed to process combined metrics ID %s: %w", cmk.ID, err)
+	// overflowService should be counted for metrics, but not logged since they only happen on
+	// service overflow.
+	addOverflow(overflowBucketName, cm.OverflowServices, 0, 0, 0)
```

Comment:
[For reviewer] I didn't change the existing behavior for metrics, but for logging, it doesn't make sense to log about global limit / per-service limit for _other service because they are only created due to service overflow, not tx / svc tx / span overflow.